core/src/main/java/io/jenkins/pluginhealth/scoring/service/ScoreService.java

```diff
@@ -68,6 +68,25 @@ public Map<String, ScoreSummary> getLatestScoresSummaryMap() {
             ));
     }
 
+    @Transactional
+    public void deleteOldPluginScores() {
+        Map<String, List<Score>> scoresByPlugin = repository.findAll().stream.collect(Collectors.groupingBy(score -> score.getPlugin().getName(),Collectors.toList
+        ));
+
+        for(List<Score> scores = scoresByPlugin.values()) {
+            if(scores.size() > 5) {
+                List<Score> scoresToRemove = scores.stream.sorted((s1, s2) -> s2.getComputedAt().compareTo(s1.getComputedAt())).skip(5).collect(Collectors.toList());
+
+              repository.deleteAll(scoresToRemove);
+            }
+        }
+    }
+
+     @Scheduled(cron = "0 0 * * * *")
+    public void scheduledDeleteOldPluginScores() {
+        deleteOldPluginScores();
+    }
```

Comment:
I'd prefer this to be on a separate class, most probably in `war` module.

Reply:
where should it be placed in between schedule or service folder in war module, most probably schedule folder

Reply:
it can be in the scoring scheduler yes.